.github/workflows/ci-pullrequest-main.yml

```diff
@@ -0,0 +1,90 @@
+name: Mock Classifier CI Pipeline
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  BuildTest:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: 6.0.x
+        
+    - name: Dotnet Restore Tools
+      run: dotnet tool restore --tool-manifest "src/.config/dotnet-tools.json"
+        
+    - name: Dotnet Restore
+      run: dotnet restore src/*.sln
+        
+    - name: Dotnet Format
+      run: dotnet format src/*.sln --no-restore
+      
+    - name: Dotnet Build
+      run: dotnet build src/*.sln --no-restore
+      
+    - name: Dotnet Publish
+      working-directory: src
+      run: dotnet publish *.sln --no-restore
+     
+    - name: Dotnet Test (unit tests only)
+      run: dotnet test /p:ExcludeByFile="**/program.cs" src/*.UnitTests/*.csproj --no-build --verbosity normal --collect:"XPlat Code Coverage"
+      
+    - name: Copy Coverage output to root
+      run: cp src/*.UnitTests/TestResults/**/coverage.cobertura.xml coverage.cobertura.xml
+      
+    - name: Code Coverage Summary
+      uses: irongut/CodeCoverageSummary@v1.2.0
```

Comment:
Perhaps using commit SHA on external action instead of tag is safer. https://michaelheap.com/ensure-github-actions-pinned-sha/